web/api/v1/api_test.go

```diff
@@ -1012,10 +1013,188 @@ func TestReadEndpoint(t *testing.T) {
 				Samples: []prompb.Sample{{Value: 1, Timestamp: 0}},
 			},
 		},
+	}, resp.Results[0])
+}
+
+func TestStreamReadEndpoint(t *testing.T) {
+	// 3 series each series.
+	// First with 119 samples. We expect 1 frame with 1 chunk.
+	// Second with 121 samples, We expect 1 frame with 2 chunks.
+	// Third with 241 samples. We expect 1 frame with 2 chunks, and 1 frame with 1 chunk for the same series due to bytes limit.
+	suite, err := promql.NewTest(t, `
+		load 1m
+			test_metric1{foo="bar1",baz="qux"} 0+100x119
+            test_metric1{foo="bar2",baz="qux"} 0+100x120
+            test_metric1{foo="bar3",baz="qux"} 0+100x240
+	`)
+	testutil.Ok(t, err)
+
+	defer suite.Close()
+
+	testutil.Ok(t, suite.Run())
+
+	api := &API{
+		Queryable:   suite.Storage(),
+		QueryEngine: suite.QueryEngine(),
+		config: func() config.Config {
+			return config.Config{
+				GlobalConfig: config.GlobalConfig{
+					ExternalLabels: labels.Labels{
+						{Name: "baz", Value: "a"},
```

Comment:
why are you adding this external label when it is already part of `test_metric1{foo="bar3",baz="qux"} 0+100x240` ?

Reply:
to check if external labels are handled properly

Reply:
properly here means that adding an external label is ignored in this case?
Is this documented somewhere so I read about it? 
Maybe add a comment here as well.

Reply:
Added comment